packages/2d/src/components/Icon.ts

```diff
@@ -0,0 +1,152 @@
+import {Image, ImageProps} from '../components/Image';
+import {computed, initial, signal} from '../decorators';
+import {
+  DependencyContext,
+  SignalValue,
+  SimpleSignal,
+} from '@motion-canvas/core/lib/signals';
+import {useLogger} from '@motion-canvas/core/lib/utils';
+
+export interface IconProps extends ImageProps {
+  /**
+   * Expects an Icon Identifier.
+   * You can find identifiers on [Icônes](https://icones.js.org/collection/all).
+   * They can look like this:
+   * * mdi:language-typescript
+   * * ph:anchor-simple-bold
+   * * ph:activity-bold
+   */
+  icon: SignalValue<string>;
+
+  /**
+   * Color of the Icon. Defaults to white.
+   * Provide the color in one of the following formats:
+   * * named color like `red`, `darkgray`, …
+   * * hexadecimal string, with or without # `#bada55`,`141414`
+   *
+   * Values can be either RGB or RGBA: `#bada55` or `#bada55aa` (latter is partially transparent)
+   */
+  color?: SignalValue<string>;
+}
+
+/**
+ * An Icon Component that provides an easy access to over 150k icons.
+ * See https://icones.js.org for all available Icons.
+ */
+export class Icon extends Image {
+  /**
+   * Internal cache
+   */
+  private static iconContentsPool: Record<string, HTMLImageElement> = {};
+
+  /**
+   * Expects an Icon Identifier.
+   * You can find identifiers on [Icônes](https://icones.js.org).
+   * They can look like this:
+   * * mdi:language-typescript
+   * * ph:anchor-simple-bold
+   * * ph:activity-bold
+   */
+  @signal()
+  public declare icon: SimpleSignal<string, this>;
+
+  /**
+   * Color of the Icon. Defaults to white.
+   * Provide the color in one of the following formats:
+   * * named color like `red`, `darkgray`, …
+   * * hexadecimal string, with or without # `#bada55`,`141414`
+   *
+   * Values can be either RGB or RGBA: `#bada55` or `#bada55aa` (latter is partially transparent)
+   */
+  @initial('white')
+  @signal()
+  public declare color: SimpleSignal<string, this>;
+
+  public constructor(props: IconProps) {
+    super(props);
+  }
+
+  /**
+   * This creates the URL that will be used as the Image source
+   * @returns Address to Iconify API for the requested Icon.
+   */
+  @computed()
+  protected svgUrl(): string {
```

Comment:
Instead of overriding the `image` method and creating this new computed method, you should be able to simply override the getter and setter for `src`:
```ts
protected getSrc(): string {
  // return the url to svg
}

protected setSrc() {
  // ignore the setter
}
```

Reply:
Yup. This does work. It's just that my IDE (VSCode) doesn't consider this to an override.

Reply:
Awesome!
> It's just that my IDE (VSCode) doesn't consider this to an override.

Signal setters and getters are magic methods so unfortunately no IDE will mark them correctly. But whenever there's a signal property, you can implement `get[Property]` and `set[Property]` methods that will override the signals default behavior.